vxlan/fdb_utils.go

```diff
@@ -18,8 +18,10 @@ func getCurrentFDBEntries(link netlink.Link) (map[string]*netlink.Neigh, error)
 
 	fdbEntries := make(map[string]*netlink.Neigh)
 	for index, n := range neighs {
-		log.Debugf("getCurrentFDBEntries: Neigh %+v", n)
-		fdbEntries[n.IP.To4().String()] = &neighs[index]
+		if n.IP != nil {
+			log.Debugf("getCurrentFDBEntries: Neigh %+v", n)
+			fdbEntries[n.IP.To4().String()] = &neighs[index]
```

Comment:
I don't understand how the problem gets fixed. 

I think if `n.IP` is nil, then `n.IP.To4().String()` should exploded with NPE exception?

Reply:
@yasker 
see this: https://play.golang.org/p/6VH-6wq8tx

Reply:
https://golang.org/pkg/net/#IP

`type IP []byte`

It turns out you can call functions of nil array in Go. Interesting.

Reply:
This is even more interesting. 

It turns out `nil` can be used as a key in `map[string]xxx`.

https://play.golang.org/p/W4HRBNrvy8